### PR TITLE
[opencv.js benchmark] add performance tests building option

### LINKED ISSUE #266: Enable performance test building option
Add an extra building option for opencv.js performance test.

### PR DESCRIPTION
**Pull request by Wenzhao-Xiang**
fix #266 

### This pullrequest changes
Add performance tests building option `--build_perf`. If building with this flag, the `/perf` folder will be copied to `bin/perf`.

### CODE REVIEW

File: modules/js/CMakeLists.txt [hunk 1]
```diff
@@ -147,3 +147,29 @@ list(APPEND opencv_test_js_file_deps "${test_data_path}" "${opencv_test_js_bin_d
...
 add_custom_target(${PROJECT_NAME}_test ALL
                   DEPENDS ${OCV_JS_PATH} ${opencv_test_js_file_deps})
+
+# perf
+set(opencv_perf_js_bin_dir "${EXECUTABLE_OUTPUT_PATH}/perf")
+set(perf_dir ${CMAKE_CURRENT_SOURCE_DIR}/perf)
+
+set(opencv_perf_js_file_deps "")
+
+# message(STATUS "${opencv_perf_js_bin_dir}")
```
Comment (huningxin) [inline modules/js/CMakeLists.txt]: Remove this line if it is not useful.

File: modules/js/CMakeLists.txt [hunk 2]
```diff
@@ -147,3 +147,29 @@ list(APPEND opencv_test_js_file_deps "${test_data_path}" "${opencv_test_js_bin_d
...
+
+# make sure the build directory exists
+file(MAKE_DIRECTORY "${opencv_perf_js_bin_dir}")
+
+# gather and copy specific files for js perf
+file(GLOB_RECURSE perf_files RELATIVE "${perf_dir}" "${perf_dir}/*")
+foreach(f ${perf_files})
+  # message(STATUS "copy ${perf_dir}/${f} ${opencv_perf_js_bin_dir}/${f}")
```
Comment (huningxin) [inline modules/js/CMakeLists.txt]: ditto

**Review by huningxin — COMMENTED** [review]
Looks good with some nits.

File: modules/js/CMakeLists.txt (hunk 1 above)
Comment (Wenzhao-Xiang) [reply modules/js/CMakeLists.txt]: done

**Review by huningxin — APPROVED** [review]
Looks good. Thanks.

**Comment by huningxin** [comment]
Let me merge it so you can move forward.

File: platforms/js/build_js.py [hunk 3]
```diff
@@ -269,6 +279,11 @@ def build_doc(self):
         if check_file(opencvjs_test_path):
             log.info("OpenCV.js tests location: %s", opencvjs_test_path)
 
+    if args.build_perf:
+        opencvjs_perf_path = os.path.join(builder.build_dir, "bin", "perf", "base.js")
+        if check_file(opencvjs_perf_path):
+            log.info("OpenCV.js tests location: %s", opencvjs_test_path)
```
Comment (terfendail) [inline platforms/js/build_js.py:285]: It's not really important, but it should be `opencvjs_perf_path`

Comment (Wenzhao-Xiang) [reply platforms/js/build_js.py:285]: Thanks for review! Track it in #276

Comment (huningxin) [reply platforms/js/build_js.py:285]: Thanks @terfendail for review. I clicked the merge button too quickly. In next two weeks, I will be on vacation. @Wenzhao-Xiang , please work closely with Vitaly on the PR review. Thanks all!

Comment (Wenzhao-Xiang) [reply platforms/js/build_js.py:285]: Get it! Have a nice time for your trip!

Comment (terfendail) [reply platforms/js/build_js.py:285]: Have a good vacation!

**Pull request merged**